packages/proto-build/src/code_generator.rs

```diff
@@ -0,0 +1,251 @@
+        }
+    }
+
+    fn prepare_dir(&self) {
+        if self.tmp_build_dir.exists() {
+            remove_dir_all(self.tmp_build_dir.clone()).unwrap();
+        }
+        create_dir_all(self.tmp_namespaced_dir()).unwrap();
+        output_version_file(
+            &self.project.name,
+            &self.project.version,
+            &self.tmp_namespaced_dir(),
+        );
+    }
+
+    fn generate_mod_file(&self) {
+        mod_gen::generate_mod_file(&self.absolute_out_dir());
+    }
+
+    fn transform(&self) {
+        transform::copy_and_transform_all(
+            &self.tmp_namespaced_dir(),
+            &self.absolute_out_dir(),
+            &self.file_descriptor_set(),
+        );
+    }
+
+    fn absolute_out_dir(&self) -> PathBuf {
+        self.root.join(&self.out_dir)
+    }
+
+    fn fmt(&self) {
+        let manifest_path = find_cargo_toml(&self.absolute_out_dir());
+        let exit_status = Command::new("cargo")
+            .arg("fmt")
+            .arg("--manifest-path")
+            .arg(manifest_path.to_string_lossy().to_string())
+            .spawn()
+            .unwrap()
+            .wait()
+            .unwrap();
+
+        if !exit_status.success() {
+            panic!(
+                "unable to format with: cargo fmt --manifest-path {}",
+                manifest_path.to_string_lossy()
+            );
+        }
+    }
+
+    fn compile_proto(&self) {
+        let include_paths = ["proto", "third_party/proto"];
+
+        let deps_dirs = self
+            .deps
+            .iter()
+            .map(|dep| self.root.join(&dep.project_dir))
+            .collect();
+        let project_dir = self.root.join(&self.project.project_dir);
+
+        let proto_includes_path = vec![deps_dirs, vec![project_dir.clone()]].concat();
+        let proto_includes_paths = proto_includes_path
+            .iter()
+            .flat_map(|dir| include_paths.iter().map(|path| dir.join(path)));
+
+        // List available paths for dependencies
+        let includes: Vec<PathBuf> = proto_includes_paths.map(PathBuf::from).collect();
+
+        let proto_paths = fs::read_dir(project_dir.join(format!("proto/{}", self.project.name)))
+            .unwrap()
+            .map(|d| d.unwrap().path().to_string_lossy().to_string())
+            .collect::<Vec<String>>();
+
+        // List available proto files
+        let mut protos: Vec<PathBuf> = vec![];
+        collect_protos(&proto_paths, &mut protos);
+
+        info!(
+            "🧪 [{}] Compiling types from protobuf definitions...",
+            self.project.name
+        );
+
+        let descriptor_file = self.tmp_namespaced_dir().join(DESCRIPTOR_FILE);
+        self.tonic_build_config
+            .clone()
+            .out_dir(self.tmp_namespaced_dir())
+            .file_descriptor_set_path(&descriptor_file)
+            .compile_well_known_types(true)
+            .compile(&protos, &includes)
+            .unwrap();
+
+        // Generate proto json (de)serialization methods
```

Comment:
pbjson_build will generates customize (de)serialize method for generated types. The default (de)serialize parse types in a wrong way, i.e, `u64`, `enum types`  See the issue:
https://github.com/tokio-rs/prost/issues/75#issuecomment-774201952